src/index.js

```diff
@@ -22,25 +22,30 @@ const getOutput = (changes, domains) => {
 
 (async () => {
     const {pull_request} = github.context.payload;
-
-    if (!pull_request) {
-        core.error('Only pull requests events can trigger this action');
-    }
-
     const domainsString = core.getInput('domains', {required: true});
     const domains = JSON.parse(domainsString);
 
     const token = core.getInput('token', {required: true});
     const octokit = github.getOctokit(token);
 
-    const response = await octokit.paginate(octokit.rest.pulls.listFiles.endpoint.merge({
-        ...github.context.repo,
-        pull_number: pull_request.number,
-    }));
-
-    const output = getOutput(response.map(({filename}) => filename), domains);
+    let files = [];
+
+    if (pull_request) {
+        files = await octokit.paginate(octokit.rest.pulls.listFiles.endpoint.merge({
+            ...github.context.repo,
+            pull_number: pull_request.number,
+        }));
+    } else {
```

Comment:
Should be fine in general since usage is specifically for our code but there are more events then `push` or `pull-request` which may occur.
A safer approach might be using switch
```
    switch (context.eventName) {
        case 'pull_request': {
            ...
        }

        case 'push': {
            ...
        }

        default: {
            core.error('Only pull requests or pushe vents can trigger this action');
        }
```

Reply:
makes sense